### PR TITLE
Move raw syntax validation from gyb to code gen

### PR DESCRIPTION
**Pull request by kimdv**
The last gym file 🥳

The removal of gym files will come in a follow up PR when this is merged

### CODE REVIEW

**Review by ahoppen — APPROVED** [review]
Thanks. This looks good to me.

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by kimdv** [comment]
Haven't run the code gen yet.
I'm still not able to do multi string literals

**Comment by ahoppen** [comment]
Ah, I thought you did now because the generated file changed. Let me know when the multi-line string literal issue is fixed and this PR is ready for review.

**Comment by kimdv** [comment]
@swift-ci please test

**Comment by kimdv** [comment]
@swift-ci please test

**Comment by kimdv** [comment]
I think it's ready now @ahoppen 🥳

**Comment by ahoppen** [comment]
@swift-ci Please test macOS

File: Sources/SwiftSyntax/generated/raw/RawSyntaxValidation.swift [hunk 1]
```diff
@@ -2644,11 +2613,8 @@ func validateLayout(layout: RawSyntaxBuffer, as kind: SyntaxKind) {
...
     assertAnyHasNoError(kind, 3, [
-      verify(layout[3], as: RawSyntax.self),
-      verify(layout[3], as: RawSyntax.self),
-    ])
+        verify(layout[3], as: RawSyntax.self)])
     assertNoError(kind, 4, verify(layout[4], as: RawUnexpectedNodesSyntax?.self))
     break
-  }
-#endif
+  }#endif
```
Comment (ahoppen) [inline Sources/SwiftSyntax/generated/raw/RawSyntaxValidation.swift:2619]: I’m surprised this is even syntactically valid. Do you think you could try to put `#endif` on its own line in a follow-up PR?

File: CodeGeneration/Sources/generate-swiftsyntax/templates/swiftsyntax/RawSyntaxValidationFile.swift [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+                  let nonNilErrors = errors.compactMap({ $0 })
+                  if nonNilErrors.count == errors.count, let firstError = nonNilErrors.first {
+                    let (file, line) = firstError.fileAndLine
+                    assertionFailure("""
+                      Error validating child at index \(index) of \(nodeKind):
+                      Node did not satisfy any node choice requirement.
+                      Validation failures:
+                      \(nonNilErrors.map({ "- \($0.description)" }).joined(separator: "\n"))
+                      """, file: file, line: line)
+                    _ = 1
```
Comment (ahoppen) [inline CodeGeneration/Sources/generate-swiftsyntax/templates/swiftsyntax/RawSyntaxValidationFile.swift:114]: Actually you can remove these `_ = 1` lines. I added them at some point during development to set a breakpoint on them and just forgot to remove them. Doesn’t need to be this PR though.

File: CodeGeneration/Sources/generate-swiftsyntax/templates/swiftsyntax/RawSyntaxValidationFile.swift [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+                            }
+                          }
+                          ExprSyntax("assertAnyHasNoError(kind, index, \(verifiedChoices))")
+                        } else {
+                          ExprSyntax("assertNoError(kind, index, verify(element, as: Raw\(node.collectionElementType.buildable).self))")
+                        }
+                      }
+                    }
+
+                    BreakStmtSyntax()
```
Comment (ahoppen) [inline CodeGeneration/Sources/generate-swiftsyntax/templates/swiftsyntax/RawSyntaxValidationFile.swift:166]: Again, not part of this PR but I don’t think we need the `break`. It’s probably a leftover of someone writing too much C++ before writing `RawSyntaxValidation.swift`

**Review by ahoppen — APPROVED** [review]
Thank you. This looks good to me. Just a few comments inline that I noticed but that aren’t really part of this PR.
